content/en/docs/reference/access-authn-authz/authentication.md

```diff
@@ -882,11 +882,20 @@ users:
         On Fedora: dnf install example-client-go-exec-plugin
 
         ...
+
+      # Whether or not to provide cluster information, which could potentially contain
+      # very large CA data, to this exec plugin as a part of the KUBERNETES_EXEC_INFO
+      # environment variable.
+      provideClusterInfo: true
 clusters:
 - name: my-cluster
   cluster:
     server: "https://172.17.4.100:6443"
     certificate-authority: "/etc/kubernetes/ca.pem"
+    extensions:
+    - name: client.authentication.k8s.io/exec # reserved extension name for per cluster exec config
+      extension:
+        some-config-per-cluster: config-data  # arbitrary config
```

Comment:
just curious, this can be a JSON representation of kind `client.authentication.k8s.io/v1beta1.ExecCredential`, and it can be something else?
Better provide a sample that gives users some ideas about this.

Reply:
When you say "this", do you mean "this extension `some-config-per-cluster: config-data`"? The "extension" data is an arbitrary JSON object. It does not have to be an `ExecCredential`. It can be anything that the user wants it to be, hence the `# arbitrary config` comment. Do you have a suggestion for how we could make this more clear?

Reply:
How about making an example JSON object that looks very much not Kubernetes-like?
`["you","can","put","anything","here"]`?

Reply:
OK, so something like this?
```yaml
...
clusters:
- name: my-cluster
  cluster:
    server: "https://172.17.4.100:6443"
    certificate-authority: "/etc/kubernetes/ca.pem"
    extensions:
    - name: client.authentication.k8s.io/exec # reserved extension name for per cluster exec config
      extension:
        arbitrary: config
        this: can be provided via the KUBERNETES_EXEC_INFO environment variable upon setting provideClusterInfo
        you: ["can", "put", "anything", "here"]
...
```

Reply:
Cool. Thanks for the feedback y'all. I tried this in c855d5d68cf05f2fee08db8030ef439ad8fbafa4. Let me know what you think.